src/jarabe/frame/clipboardicon.py

```diff
@@ -176,11 +178,27 @@ def show_notification(self):
             self._notif_icon, Gtk.CornerType.BOTTOM_LEFT)
         self._notif_icon.connect('drag_data_get', self._drag_data_get_cb)
         self._notif_icon.connect('drag-begin', self._drag_begin_cb)
+        self._notif_icon.connect('drag-end', self._drag_end_cb)
         self._notif_icon.drag_source_set(Gdk.ModifierType.BUTTON1_MASK,
                                          self._get_targets(),
                                          Gdk.DragAction.COPY)
 
+    def _drag_end_cb(self, widget, context):
+        '''
+        If we are dropping onto the journal, only then we save the object in the journal,
+        the destination window (Gdk.X11Window) of the journal has the ID >= 8388800
+        '''
+        logging.debug(context.get_dest_window().get_xid())
+        print (context.get_dest_window().get_xid())
+        if context.get_dest_window()!= None and self.drag_inprocess == True and \
+                context.get_dest_window().get_xid()>=8388800 and context.get_dest_window().get_xid()<=8389000:
```

Comment:
You should not try and detect when a clipboard item is dropped on the journal.  You need to only change the code for the journal activity icon in the frame.

The journal icon should deal with getting the drag data and saving it.  See how the clipboard tray does it with the drag-data-received signal:  https://github.com/sugarlabs/sugar/blob/master/src/jarabe/frame/clipboardtray.py#L173